boto/route53/connection.py

```diff
@@ -136,6 +136,48 @@ def get_all_hosted_zones(self, start_marker=None, zone_list=None):
             e = self.get_all_hosted_zones(next_marker, zone_list)
         return e
 
+    def get_all_hosted_zones_by_name(self, dns_name=None, hosted_zone_id=None,
+                                     zone_list=None):
+        """
+        Returns a Python data structure with information about all
+        Hosted Zones defined for the AWS account.
+
+        :param string dns_name: Specifies the first dnsname to be returned in
+            the list of domains. Mostly meant to be used with hosted_zone_id.
+            If left blank, will return all dns records.
+        :param string hosted_zone_id: Should not be included in first query
+            to get_all_hosted_zones_by_name, but should be used, along with
+            dns_name when dealing with truncated lists.
+        :param list zone_list: a HostedZones list to prepend to results
```

Comment:
Cool - updated, as well as updated the get_all_hosted_zones method, as that's what I based my docstring off of.